sycl/source/detail/scheduler/commands.cpp

```diff
@@ -402,8 +418,27 @@ class DispatchHostTask {
         interop_handle IH{MReqToMem, HostTask.MQueue,
                           HostTask.MQueue->getDeviceImplPtr(),
                           HostTask.MQueue->getContextImplPtr()};
-
-        HostTask.MHostTask->call(MThisCmd->MEvent->getHostProfilingInfo(), IH);
+        if (IH.get_backend() == backend::ext_oneapi_cuda ||
+            IH.get_backend() == backend::ext_oneapi_hip) {
```

Comment:
I would have preferred that the UR adapter would tell us if we could use this path, instead of making it the runtime checking the type of backend. That said, I can live with this as a stop-gap solution. Is the plan to have the other backends support this UR functionality as well?

Reply:
We can query this info from UR actually, but I think it's better only to use this entry point if it's needed for correctness, rather than if the entry point is supported. Although I don't feel too strongly about this either way, I can change this to query whether the device supports the experimental UR entry point.

There are indeed plans for other backends to support this although we are focussing our energy on CUDA and HIP for now, since interop is a big perf limiter for these backends. Interop isn't a big issue for other backends at the moment, but I think this will soon be supported in other backends as well.

Reply:
Maybe as an interim solution I'll add a TODO in code, saying `should this entry point be used for all the backends that support it?`. This is an open question.

Reply:
I am fine with a TODO, but in the end I would still prefer if we could rely on the support query for this. Hope would be that any backend that reports support for it should be able to do it efficiently and correctly, as I assume we don't want different behavior for the API.

Maybe it means we can eventually always use this path and scrap some of the implementation in our library. 😉

Reply:
Yeah I think that's a good idea! I will change actually to use the query, if that is the preferable path

Reply:
I have updated the check to query `piDeviceGetInfo` to see if the entry point is supported.